tests/suites/test_suite_ccm.function

```diff
@@ -1,5 +1,67 @@
 /* BEGIN_HEADER */
```

Comment:
Missing: tests that skip calling `update_ad` when the AD is empty, that skip calling `update` when the plaintext/ciphertext is empty, or both.